### DISCUSSION TITLE
Prevent Infinite Loops on User Context

### DISCUSSION THREAD

**Discussion by troy-whitespark** [spatie/flareapp.io-roadmap#35] [Ideas]
Currently, the flare app automatically serializes the `User` model (which is certainly helpful).

However, if there is complexity in the fields of a `User`, an infinite loop can occur. Consider the following example pseudocode that just happened in our environment:

```
class User {
    protected $appends [ 'latest_invoice_amount'];
    getLatestInvoiceAmountAttribute() {
        try { 
            return $user->getDataFromStripe();
         } catch (StripeException $exception) {
            Flare::report($exception);
            return 0;
        }
    }
}
```

We try to get notified in the case where everything goes wrong... which causes everything to go much _worse_. If you report a flare error in any User method that's called upon serialization, you've got an infinite loop which breaks the whole app simply because you tried to _report_ something went wrong.

Yes, I'm aware we can configure the User's `->toFlare()` method and serialize less (and we did that). I suggest that the error handler should come with some rudimentary checks to ensure the error handler itself is not _causing_ the problem.

Thanks!

**Comment by AlexVanderbist** [comment]
Thanks for reporting this. I'll move this issue to the Ignition repository

**Reply by AlexVanderbist** [reply]
ETA would be around the same time as the Laravel 9 release!
